graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -442,13 +435,16 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
             # Make sure there's a marked location right before it for the optional Backtrack
             # to jump back to. Otherwise, the traversal could rewind to an old marked location
             # and might ignore entire stretches of applied filtering.
-            if context['marked_location_stack'][-1].num_traverses > 0:
+            # The only way there might not be a marked location here is if the current location
+            # has existing child locations that are not folds.
```

Comment:
Can you document this assumption? We don't have a good way to deal with these, but a convention like this might be ok for now: # Assumption: If the current location doesn't have existing child locations that are not folds, then there is no mark location here.

This way it's clerer that something is assumed, you can name your assumptions with short names if you want to, or if we do have a better way in the future we can easily find all of them.